### ISSUE TITLE
Enhance placements to support staged rollouts

### ISSUE THREAD

**Issue by prateek** [m3db/m3cluster#140]
We currently optimise placement generation/modification logic to maximise randomness when distributing shards, so as to evenly distribute load. This gets in the way of our ability to rollout new versions to a single replica. 

Put another way, we run M3DB with replication factor 3. To minimise risk during rollout, we'd like the ability to rollout to a single node in each "replica set". We should enhance our placement algorithm to support this.

**Comment by xichen2020** [comment]
Sounds like what you want is the mirrored placement, which the new agg tier also needs. @cw9 has made good progress on that front so I'll let him to chime in here.

**Comment by prateek** [comment]
Yeah it's kinda like that, dunno what approach you've taken for it. I was thinking we can accomplish this without any interface changes to the placement, only implementation of the placement algorithm.
